util/fipstools/inject_hash/macho_parser/macho_parser.c

```diff
@@ -0,0 +1,165 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <stdint.h>
+
+#include "common.h"
+#include "macho_parser.h"
+
+// Documentation for the Mach-O structs can be found in macho-o/loader.h and mach-o/nlist.h
+int read_macho_file(const char *filename, machofile *macho) {
+    FILE *file = NULL;
+    load_cmd *load_commands = NULL;
+    int ret = 1;
+
+    file = fopen(filename, "rb");
+    if (!file) {
+        LOG_ERROR("Error opening file %s", filename);
+        ret = 0;
+        goto end;
+    }
+
+    fread(&macho->macho_header, sizeof(macho_header), 1, file);
+    if(macho->macho_header.magic != MH_MAGIC_64) {
+        LOG_ERROR("File is not a 64-bit Mach-O file");
+        ret = 0;
+        goto end;
+    }
+
+    load_commands = malloc(macho->macho_header.sizeofcmds);
+    fread(load_commands, macho->macho_header.sizeofcmds, 1, file);
+
+    // We're only looking for __text, __const in the __TEXT segment, and the string & symbol tables
+    macho->num_sections = 4;
+    macho->sections = malloc(macho->num_sections * sizeof(section_info));
+
+    uint32_t section_index = 0;
+    for (uint32_t i = 0; i < macho->macho_header.sizeofcmds / BIT_MODIFIER; i += load_commands[i].cmdsize / BIT_MODIFIER) {
+        if (load_commands[i].cmd == LC_SEG) {
+            segment_load_cmd *segment = (segment_load_cmd *)&load_commands[i];
+            if (strcmp(segment->segname, "__TEXT") == 0) {
+                section_data *sections = (section_data *)&segment[1];
+                for (uint32_t j = 0; j < segment->nsects; j++) {
+                    if (strcmp(sections[j].sectname, "__text") == 0 || strcmp(sections[j].sectname, "__const") == 0) {
```

Comment:
Instead of incrementing a `section_index` counter, why not just have an explicit index at which "__text", "__const", etc. are stored?

Reply:
I think the current implementation is more easily expandable if we ever decide that we're interested in more than just the `__text` and `__const` sections, and I'm unsure if changing it around provides any added benefit to our current use-case.

Reply:
I think having a known location for each section could make it more robust to potential(?) changes in the order that sections are processed. I don't see it limiting  expansion to handle other sections; sections being added to the processing would likewise be assigned to a specific index.